hypothesis-python/src/hypothesis/searchstrategy/types.py

```diff
@@ -185,6 +187,10 @@ def from_typing_type(thing):
         typing.io.BinaryIO: st.builds(io.BytesIO, st.binary()),  # type: ignore
         typing.io.TextIO: st.builds(io.StringIO, st.text()),  # type: ignore
         typing.Reversible: st.lists(st.integers()),
+        typing.Collection: st.lists(st.integers()),
```

Comment:
It's actually incorrect to resolve generic types from the typing module this way - `lists(integers)` will *not* generate instances of `Collection[Text]`, for example.

The (sadly much more complicated) code to handle generic types is further down in this module and doesn't need to be touched for this PR.